src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -184,39 +186,143 @@ public async Task EmissionsForecastsBatch_MissingRequiredParams_ReturnsBadReques
         Assert.That(result?.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
     }
 
-    [Test]
-    public async Task EmissionsForecastsBatch_SupportedDataSources_ReturnsOk()
+    [TestCase("2021-09-01T08:30:00Z", "2021-09-01T09:00:00Z", "2021-09-01T12:00:00Z", "eastus", 1, TestName = "EmissionsForecastsBatch expects OK for eastus 1 element")]
+    [TestCase("2021-09-01T08:30:00Z", "2021-09-01T08:30:00Z", "2021-09-02T08:30:00Z", "westus", 3, TestName = "EmissionsForecastsBatch expects OK for westus 3 element")]
+    public async Task EmissionsForecastsBatch_SupportedDataSources_ReturnsOk(DateTimeOffset reqAt, DateTimeOffset start, DateTimeOffset end, string location, int nelems)
     {
         IgnoreTestForDataSource("data source does not implement '/emissions/forecasts/batch'", DataSourceType.JSON);
 
         _dataSourceMocker.SetupBatchForecastMock();
+        var inputData = Enumerable.Range(0, nelems).Select(x => new EmissionsForecastBatchDTO() 
```

Comment:
Similar to above.  Prefer creating a JSON object that mimics what a user would send who can't call `new EmissionsForecastBatchDTO() `